src/braket/ahs/analog_hamiltonian_simulation.py

```diff
@@ -63,6 +71,93 @@ def _hamiltonian_to_ir(self) -> ir.Hamiltonian:
             drivingFields=terms["driving_fields"], shiftingFields=terms["shifting_fields"]
         )
 
+    def discretize(self, device: AwsDevice) -> AnalogHamiltonianSimulation:
+        """ Creates a new AnalogHamiltonianSimulation with all numerical values represented
+            as Decimal objects with fixed precision based on the capabilities of the device.
+
+            Args:
+                device (AwsDevice): The device for which to discretize the program.
+
+            Returns:
+                AnalogHamiltonianSimulation: A discretized version of this program.
+
+            Raises:
+                DiscretizeError: If unable to discretize the program.
+        """
+
+        # Gather resolution values
```

Comment:
You should first validate that the device is a AHS device (with the correct action type), and throw an error otherwise

Reply:
Added validation.